### ISSUE TITLE
no-console rule does not report when console method is passed as callback

### ISSUE THREAD

**Issue by kirick13** [oxc-project/oxc#11780]
The `no-console` rule doesn’t report anything when a console method is passed as a callback, like this:
```ts
somePromise().catch(console.error);
```
It only reports when the method is actually called, like
```ts
console.error('123');
```
ESLint flags both cases. Just wanted to check if this is something that should be aligned in oxlint as well.

Oxlint version: 1.1.0

**Comment by camc314** [comment]
fixed in https://github.com/oxc-project/oxc/pull/11790
